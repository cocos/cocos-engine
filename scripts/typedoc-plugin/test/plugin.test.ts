import ps from 'path';
import { runTest } from './test-util';

test(`Test`, async () => {
    await runTest(ps.join(__dirname, 'dummy-engine'));
});
