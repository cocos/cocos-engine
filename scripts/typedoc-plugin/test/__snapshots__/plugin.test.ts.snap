// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test 1`] = `
Object {
  "categories": undefined,
  "ccCategories": Array [
    Object {
      "items": Array [
        1,
        2,
        3,
        4,
        5,
        7,
        9,
        12,
        15,
        16,
        18,
        22,
        23,
        24,
        25,
        27,
        31,
        32,
        34,
      ],
      "title": undefined,
    },
  ],
  "children": Array [
    Object {
      "categories": undefined,
      "children": undefined,
      "comment": undefined,
      "decorates": undefined,
      "decorators": undefined,
      "defaultValue": undefined,
      "extendedBy": undefined,
      "extendedTypes": undefined,
      "flags": Object {},
      "groups": undefined,
      "id": 29,
      "implementationOf": undefined,
      "implementedBy": undefined,
      "implementedTypes": undefined,
      "indexSignature": undefined,
      "inheritedFrom": undefined,
      "kind": 16777216,
      "kindString": "Reference",
      "name": "XXVec3",
      "overwrites": undefined,
      "signatures": undefined,
      "sources": Array [
        Object {
          "character": 9,
          "fileName": "core/math/index.ts",
          "line": 2,
        },
      ],
      "target": 9,
      "type": undefined,
      "typeParameter": undefined,
    },
    Object {
      "categories": undefined,
      "children": undefined,
      "comment": undefined,
      "decorates": undefined,
      "decorators": undefined,
      "defaultValue": undefined,
      "extendedBy": undefined,
      "extendedTypes": undefined,
      "flags": Object {},
      "groups": undefined,
      "id": 30,
      "implementationOf": undefined,
      "implementedBy": undefined,
      "implementedTypes": undefined,
      "indexSignature": undefined,
      "inheritedFrom": undefined,
      "kind": 16777216,
      "kindString": "Reference",
      "name": "XXVec4",
      "overwrites": undefined,
      "signatures": undefined,
      "sources": Array [
        Object {
          "character": 9,
          "fileName": "core/math/index.ts",
          "line": 3,
        },
      ],
      "target": 12,
      "type": undefined,
      "typeParameter": undefined,
    },
    Object {
      "categories": undefined,
      "children": Array [
        Object {
          "categories": undefined,
          "children": Array [
            Object {
              "categories": undefined,
              "children": undefined,
              "comment": undefined,
              "decorates": undefined,
              "decorators": undefined,
              "defaultValue": "0",
              "extendedBy": undefined,
              "extendedTypes": undefined,
              "flags": Object {},
              "groups": undefined,
              "id": 19,
              "implementationOf": undefined,
              "implementedBy": undefined,
              "implementedTypes": undefined,
              "indexSignature": undefined,
              "inheritedFrom": undefined,
              "kind": 16,
              "kindString": "Enumeration member",
              "name": "XXEnumMember1",
              "overwrites": undefined,
              "signatures": undefined,
              "sources": Array [
                Object {
                  "character": 22,
                  "fileName": "core/math/index.ts",
                  "line": 17,
                },
              ],
              "type": undefined,
              "typeParameter": undefined,
            },
            Object {
              "categories": undefined,
              "children": undefined,
              "comment": undefined,
              "decorates": undefined,
              "decorators": undefined,
              "defaultValue": "1",
              "extendedBy": undefined,
              "extendedTypes": undefined,
              "flags": Object {},
              "groups": undefined,
              "id": 20,
              "implementationOf": undefined,
              "implementedBy": undefined,
              "implementedTypes": undefined,
              "indexSignature": undefined,
              "inheritedFrom": undefined,
              "kind": 16,
              "kindString": "Enumeration member",
              "name": "XXEnumMember2",
              "overwrites": undefined,
              "signatures": undefined,
              "sources": Array [
                Object {
                  "character": 37,
                  "fileName": "core/math/index.ts",
                  "line": 17,
                },
              ],
              "type": undefined,
              "typeParameter": undefined,
            },
            Object {
              "categories": undefined,
              "children": undefined,
              "comment": undefined,
              "decorates": undefined,
              "decorators": undefined,
              "defaultValue": "2",
              "extendedBy": undefined,
              "extendedTypes": undefined,
              "flags": Object {},
              "groups": undefined,
              "id": 21,
              "implementationOf": undefined,
              "implementedBy": undefined,
              "implementedTypes": undefined,
              "indexSignature": undefined,
              "inheritedFrom": undefined,
              "kind": 16,
              "kindString": "Enumeration member",
              "name": "XXEnumMember3",
              "overwrites": undefined,
              "signatures": undefined,
              "sources": Array [
                Object {
                  "character": 52,
                  "fileName": "core/math/index.ts",
                  "line": 17,
                },
              ],
              "type": undefined,
              "typeParameter": undefined,
            },
          ],
          "comment": undefined,
          "decorates": undefined,
          "decorators": undefined,
          "defaultValue": undefined,
          "extendedBy": undefined,
          "extendedTypes": undefined,
          "flags": Object {},
          "groups": Array [
            Object {
              "children": Array [
                19,
                20,
                21,
              ],
              "kind": 16,
              "title": "Enumeration members",
            },
          ],
          "id": 18,
          "implementationOf": undefined,
          "implementedBy": undefined,
          "implementedTypes": undefined,
          "indexSignature": undefined,
          "inheritedFrom": undefined,
          "kind": 8,
          "kindString": "Enumeration",
          "name": "XXEnum",
          "overwrites": undefined,
          "signatures": undefined,
          "sources": Array [
            Object {
              "character": 12,
              "fileName": "core/math/index.ts",
              "line": 17,
            },
          ],
          "type": undefined,
          "typeParameter": undefined,
        },
        Object {
          "categories": undefined,
          "children": Array [
            Object {
              "categories": undefined,
              "children": undefined,
              "comment": undefined,
              "decorates": undefined,
              "decorators": undefined,
              "defaultValue": undefined,
              "extendedBy": undefined,
              "extendedTypes": undefined,
              "flags": Object {},
              "groups": undefined,
              "id": 10,
              "implementationOf": undefined,
              "implementedBy": undefined,
              "implementedTypes": undefined,
              "indexSignature": undefined,
              "inheritedFrom": undefined,
              "kind": 512,
              "kindString": "Constructor",
              "name": "constructor",
              "overwrites": undefined,
              "signatures": Array [
                Object {
                  "comment": undefined,
                  "decorates": undefined,
                  "decorators": undefined,
                  "flags": Object {},
                  "id": 11,
                  "implementationOf": undefined,
                  "inheritedFrom": undefined,
                  "kind": 16384,
                  "kindString": "Constructor signature",
                  "name": "new XXVec3",
                  "overwrites": undefined,
                  "parameters": undefined,
                  "type": Object {
                    "id": 9,
                    "name": "XXVec3",
                    "type": "reference",
                  },
                  "typeParameter": undefined,
                },
              ],
              "sources": undefined,
              "type": undefined,
              "typeParameter": undefined,
            },
          ],
          "comment": undefined,
          "decorates": undefined,
          "decorators": undefined,
          "defaultValue": undefined,
          "extendedBy": undefined,
          "extendedTypes": undefined,
          "flags": Object {},
          "groups": Array [
            Object {
              "children": Array [
                10,
              ],
              "kind": 512,
              "title": "Constructors",
            },
          ],
          "id": 9,
          "implementationOf": undefined,
          "implementedBy": undefined,
          "implementedTypes": undefined,
          "indexSignature": undefined,
          "inheritedFrom": undefined,
          "kind": 128,
          "kindString": "Class",
          "name": "XXVec3",
          "overwrites": undefined,
          "signatures": undefined,
          "sources": Array [
            Object {
              "character": 13,
              "fileName": "core/math/vec3.ts",
              "line": 2,
            },
          ],
          "type": undefined,
          "typeParameter": undefined,
        },
        Object {
          "categories": undefined,
          "children": Array [
            Object {
              "categories": undefined,
              "children": undefined,
              "comment": undefined,
              "decorates": undefined,
              "decorators": undefined,
              "defaultValue": undefined,
              "extendedBy": undefined,
              "extendedTypes": undefined,
              "flags": Object {},
              "groups": undefined,
              "id": 13,
              "implementationOf": undefined,
              "implementedBy": undefined,
              "implementedTypes": undefined,
              "indexSignature": undefined,
              "inheritedFrom": undefined,
              "kind": 512,
              "kindString": "Constructor",
              "name": "constructor",
              "overwrites": undefined,
              "signatures": Array [
                Object {
                  "comment": undefined,
                  "decorates": undefined,
                  "decorators": undefined,
                  "flags": Object {},
                  "id": 14,
                  "implementationOf": undefined,
                  "inheritedFrom": undefined,
                  "kind": 16384,
                  "kindString": "Constructor signature",
                  "name": "new XXVec4",
                  "overwrites": undefined,
                  "parameters": undefined,
                  "type": Object {
                    "id": 12,
                    "name": "XXVec4",
                    "type": "reference",
                  },
                  "typeParameter": undefined,
                },
              ],
              "sources": undefined,
              "type": undefined,
              "typeParameter": undefined,
            },
          ],
          "comment": undefined,
          "decorates": undefined,
          "decorators": undefined,
          "defaultValue": undefined,
          "extendedBy": undefined,
          "extendedTypes": undefined,
          "flags": Object {},
          "groups": Array [
            Object {
              "children": Array [
                13,
              ],
              "kind": 512,
              "title": "Constructors",
            },
          ],
          "id": 12,
          "implementationOf": undefined,
          "implementedBy": undefined,
          "implementedTypes": undefined,
          "indexSignature": undefined,
          "inheritedFrom": undefined,
          "kind": 128,
          "kindString": "Class",
          "name": "XXVec4",
          "overwrites": undefined,
          "signatures": undefined,
          "sources": Array [
            Object {
              "character": 13,
              "fileName": "core/math/vec4.ts",
              "line": 2,
            },
          ],
          "type": undefined,
          "typeParameter": undefined,
        },
        Object {
          "categories": undefined,
          "children": Array [
            Object {
              "categories": undefined,
              "children": undefined,
              "comment": undefined,
              "decorates": undefined,
              "decorators": undefined,
              "defaultValue": undefined,
              "extendedBy": undefined,
              "extendedTypes": undefined,
              "flags": Object {},
              "groups": undefined,
              "id": 17,
              "implementationOf": undefined,
              "implementedBy": undefined,
              "implementedTypes": undefined,
              "indexSignature": undefined,
              "inheritedFrom": undefined,
              "kind": 1024,
              "kindString": "Property",
              "name": "xxInterfaceMember1",
              "overwrites": undefined,
              "signatures": undefined,
              "sources": Array [
                Object {
                  "character": 4,
                  "fileName": "core/math/index.ts",
                  "line": 14,
                },
              ],
              "type": Object {
                "name": "number",
                "type": "intrinsic",
              },
              "typeParameter": undefined,
            },
          ],
          "comment": undefined,
          "decorates": undefined,
          "decorators": undefined,
          "defaultValue": undefined,
          "extendedBy": undefined,
          "extendedTypes": undefined,
          "flags": Object {},
          "groups": Array [
            Object {
              "children": Array [
                17,
              ],
              "kind": 1024,
              "title": "Properties",
            },
          ],
          "id": 16,
          "implementationOf": undefined,
          "implementedBy": undefined,
          "implementedTypes": undefined,
          "indexSignature": undefined,
          "inheritedFrom": undefined,
          "kind": 256,
          "kindString": "Interface",
          "name": "XXInterface",
          "overwrites": undefined,
          "signatures": undefined,
          "sources": Array [
            Object {
              "character": 17,
              "fileName": "core/math/index.ts",
              "line": 13,
            },
          ],
          "type": undefined,
          "typeParameter": undefined,
        },
        Object {
          "categories": undefined,
          "children": undefined,
          "comment": undefined,
          "decorates": undefined,
          "decorators": undefined,
          "defaultValue": "0.0",
          "extendedBy": undefined,
          "extendedTypes": undefined,
          "flags": Object {
            "isConst": true,
          },
          "groups": undefined,
          "id": 15,
          "implementationOf": undefined,
          "implementedBy": undefined,
          "implementedTypes": undefined,
          "indexSignature": undefined,
          "inheritedFrom": undefined,
          "kind": 32,
          "kindString": "Variable",
          "name": "xxVariable",
          "overwrites": undefined,
          "signatures": undefined,
          "sources": Array [
            Object {
              "character": 13,
              "fileName": "core/math/index.ts",
              "line": 5,
            },
          ],
          "type": Object {
            "type": "literal",
            "value": 0,
          },
          "typeParameter": undefined,
        },
        Object {
          "categories": undefined,
          "children": undefined,
          "comment": undefined,
          "decorates": undefined,
          "decorators": undefined,
          "defaultValue": undefined,
          "extendedBy": undefined,
          "extendedTypes": undefined,
          "flags": Object {},
          "groups": undefined,
          "id": 2,
          "implementationOf": undefined,
          "implementedBy": undefined,
          "implementedTypes": undefined,
          "indexSignature": undefined,
          "inheritedFrom": undefined,
          "kind": 64,
          "kindString": "Function",
          "name": "xxFn",
          "overwrites": undefined,
          "signatures": Array [
            Object {
              "comment": undefined,
              "decorates": undefined,
              "decorators": undefined,
              "flags": Object {},
              "id": 3,
              "implementationOf": undefined,
              "inheritedFrom": undefined,
              "kind": 4096,
              "kindString": "Call signature",
              "name": "xxFn",
              "overwrites": undefined,
              "parameters": undefined,
              "type": Object {
                "name": "void",
                "type": "intrinsic",
              },
              "typeParameter": undefined,
            },
          ],
          "sources": Array [
            Object {
              "character": 16,
              "fileName": "core/math/index.ts",
              "line": 7,
            },
          ],
          "type": undefined,
          "typeParameter": undefined,
        },
        Object {
          "categories": undefined,
          "children": undefined,
          "comment": undefined,
          "decorates": undefined,
          "decorators": undefined,
          "defaultValue": undefined,
          "extendedBy": undefined,
          "extendedTypes": undefined,
          "flags": Object {},
          "groups": undefined,
          "id": 4,
          "implementationOf": undefined,
          "implementedBy": undefined,
          "implementedTypes": undefined,
          "indexSignature": undefined,
          "inheritedFrom": undefined,
          "kind": 64,
          "kindString": "Function",
          "name": "xxFunctionWithMultiSignature",
          "overwrites": undefined,
          "signatures": Array [
            Object {
              "comment": undefined,
              "decorates": undefined,
              "decorators": undefined,
              "flags": Object {},
              "id": 5,
              "implementationOf": undefined,
              "inheritedFrom": undefined,
              "kind": 4096,
              "kindString": "Call signature",
              "name": "xxFunctionWithMultiSignature",
              "overwrites": undefined,
              "parameters": Array [
                Object {
                  "comment": undefined,
                  "decorates": undefined,
                  "decorators": undefined,
                  "defaultValue": undefined,
                  "flags": Object {},
                  "id": 6,
                  "kind": 32768,
                  "kindString": "Parameter",
                  "name": "p",
                  "type": Object {
                    "name": "string",
                    "type": "intrinsic",
                  },
                },
              ],
              "type": Object {
                "name": "void",
                "type": "intrinsic",
              },
              "typeParameter": undefined,
            },
            Object {
              "comment": undefined,
              "decorates": undefined,
              "decorators": undefined,
              "flags": Object {},
              "id": 7,
              "implementationOf": undefined,
              "inheritedFrom": undefined,
              "kind": 4096,
              "kindString": "Call signature",
              "name": "xxFunctionWithMultiSignature",
              "overwrites": undefined,
              "parameters": Array [
                Object {
                  "comment": undefined,
                  "decorates": undefined,
                  "decorators": undefined,
                  "defaultValue": undefined,
                  "flags": Object {},
                  "id": 8,
                  "kind": 32768,
                  "kindString": "Parameter",
                  "name": "p",
                  "type": Object {
                    "name": "number",
                    "type": "intrinsic",
                  },
                },
              ],
              "type": Object {
                "name": "void",
                "type": "intrinsic",
              },
              "typeParameter": undefined,
            },
          ],
          "sources": Array [
            Object {
              "character": 16,
              "fileName": "core/math/index.ts",
              "line": 9,
            },
          ],
          "type": undefined,
          "typeParameter": undefined,
        },
      ],
      "comment": undefined,
      "decorates": undefined,
      "decorators": undefined,
      "defaultValue": undefined,
      "extendedBy": undefined,
      "extendedTypes": undefined,
      "flags": Object {},
      "groups": Array [
        Object {
          "children": Array [
            18,
          ],
          "kind": 8,
          "title": "Enumerations",
        },
        Object {
          "children": Array [
            9,
            12,
          ],
          "kind": 128,
          "title": "Classes",
        },
        Object {
          "children": Array [
            16,
          ],
          "kind": 256,
          "title": "Interfaces",
        },
        Object {
          "children": Array [
            15,
          ],
          "kind": 32,
          "title": "Variables",
        },
        Object {
          "children": Array [
            2,
            4,
          ],
          "kind": 64,
          "title": "Functions",
        },
      ],
      "id": 1,
      "implementationOf": undefined,
      "implementedBy": undefined,
      "implementedTypes": undefined,
      "indexSignature": undefined,
      "inheritedFrom": undefined,
      "kind": 4,
      "kindString": "Namespace",
      "name": "math",
      "overwrites": undefined,
      "signatures": undefined,
      "sources": Array [
        Object {
          "character": 0,
          "fileName": "core/math/index.ts",
          "line": 2,
        },
      ],
      "type": undefined,
      "typeParameter": undefined,
    },
    Object {
      "categories": undefined,
      "children": Array [
        Object {
          "categories": undefined,
          "children": undefined,
          "comment": undefined,
          "decorates": undefined,
          "decorators": undefined,
          "defaultValue": "0",
          "extendedBy": undefined,
          "extendedTypes": undefined,
          "flags": Object {},
          "groups": undefined,
          "id": 35,
          "implementationOf": undefined,
          "implementedBy": undefined,
          "implementedTypes": undefined,
          "indexSignature": undefined,
          "inheritedFrom": undefined,
          "kind": 16,
          "kindString": "Enumeration member",
          "name": "XXEnumMember1",
          "overwrites": undefined,
          "signatures": undefined,
          "sources": Array [
            Object {
              "character": 22,
              "fileName": "core/math/index.ts",
              "line": 17,
            },
          ],
          "type": undefined,
          "typeParameter": undefined,
        },
        Object {
          "categories": undefined,
          "children": undefined,
          "comment": undefined,
          "decorates": undefined,
          "decorators": undefined,
          "defaultValue": "1",
          "extendedBy": undefined,
          "extendedTypes": undefined,
          "flags": Object {},
          "groups": undefined,
          "id": 36,
          "implementationOf": undefined,
          "implementedBy": undefined,
          "implementedTypes": undefined,
          "indexSignature": undefined,
          "inheritedFrom": undefined,
          "kind": 16,
          "kindString": "Enumeration member",
          "name": "XXEnumMember2",
          "overwrites": undefined,
          "signatures": undefined,
          "sources": Array [
            Object {
              "character": 37,
              "fileName": "core/math/index.ts",
              "line": 17,
            },
          ],
          "type": undefined,
          "typeParameter": undefined,
        },
        Object {
          "categories": undefined,
          "children": undefined,
          "comment": undefined,
          "decorates": undefined,
          "decorators": undefined,
          "defaultValue": "2",
          "extendedBy": undefined,
          "extendedTypes": undefined,
          "flags": Object {},
          "groups": undefined,
          "id": 37,
          "implementationOf": undefined,
          "implementedBy": undefined,
          "implementedTypes": undefined,
          "indexSignature": undefined,
          "inheritedFrom": undefined,
          "kind": 16,
          "kindString": "Enumeration member",
          "name": "XXEnumMember3",
          "overwrites": undefined,
          "signatures": undefined,
          "sources": Array [
            Object {
              "character": 52,
              "fileName": "core/math/index.ts",
              "line": 17,
            },
          ],
          "type": undefined,
          "typeParameter": undefined,
        },
      ],
      "comment": undefined,
      "decorates": undefined,
      "decorators": undefined,
      "defaultValue": undefined,
      "extendedBy": undefined,
      "extendedTypes": undefined,
      "flags": Object {},
      "groups": Array [
        Object {
          "children": Array [
            35,
            36,
            37,
          ],
          "kind": 16,
          "title": "Enumeration members",
        },
      ],
      "id": 34,
      "implementationOf": undefined,
      "implementedBy": undefined,
      "implementedTypes": undefined,
      "indexSignature": undefined,
      "inheritedFrom": undefined,
      "kind": 8,
      "kindString": "Enumeration",
      "name": "XXEnum",
      "overwrites": undefined,
      "signatures": undefined,
      "sources": Array [
        Object {
          "character": 12,
          "fileName": "core/math/index.ts",
          "line": 17,
        },
      ],
      "type": undefined,
      "typeParameter": undefined,
    },
    Object {
      "categories": undefined,
      "children": Array [
        Object {
          "categories": undefined,
          "children": undefined,
          "comment": undefined,
          "decorates": undefined,
          "decorators": undefined,
          "defaultValue": undefined,
          "extendedBy": undefined,
          "extendedTypes": undefined,
          "flags": Object {},
          "groups": undefined,
          "id": 33,
          "implementationOf": undefined,
          "implementedBy": undefined,
          "implementedTypes": undefined,
          "indexSignature": undefined,
          "inheritedFrom": undefined,
          "kind": 1024,
          "kindString": "Property",
          "name": "xxInterfaceMember1",
          "overwrites": undefined,
          "signatures": undefined,
          "sources": Array [
            Object {
              "character": 4,
              "fileName": "core/math/index.ts",
              "line": 14,
            },
          ],
          "type": Object {
            "name": "number",
            "type": "intrinsic",
          },
          "typeParameter": undefined,
        },
      ],
      "comment": undefined,
      "decorates": undefined,
      "decorators": undefined,
      "defaultValue": undefined,
      "extendedBy": undefined,
      "extendedTypes": undefined,
      "flags": Object {},
      "groups": Array [
        Object {
          "children": Array [
            33,
          ],
          "kind": 1024,
          "title": "Properties",
        },
      ],
      "id": 32,
      "implementationOf": undefined,
      "implementedBy": undefined,
      "implementedTypes": undefined,
      "indexSignature": undefined,
      "inheritedFrom": undefined,
      "kind": 256,
      "kindString": "Interface",
      "name": "XXInterface",
      "overwrites": undefined,
      "signatures": undefined,
      "sources": Array [
        Object {
          "character": 17,
          "fileName": "core/math/index.ts",
          "line": 13,
        },
      ],
      "type": undefined,
      "typeParameter": undefined,
    },
    Object {
      "categories": undefined,
      "children": undefined,
      "comment": undefined,
      "decorates": undefined,
      "decorators": undefined,
      "defaultValue": "0.0",
      "extendedBy": undefined,
      "extendedTypes": undefined,
      "flags": Object {
        "isConst": true,
      },
      "groups": undefined,
      "id": 31,
      "implementationOf": undefined,
      "implementedBy": undefined,
      "implementedTypes": undefined,
      "indexSignature": undefined,
      "inheritedFrom": undefined,
      "kind": 32,
      "kindString": "Variable",
      "name": "xxVariable",
      "overwrites": undefined,
      "signatures": undefined,
      "sources": Array [
        Object {
          "character": 13,
          "fileName": "core/math/index.ts",
          "line": 5,
        },
      ],
      "type": Object {
        "type": "literal",
        "value": 0,
      },
      "typeParameter": undefined,
    },
    Object {
      "categories": undefined,
      "children": undefined,
      "comment": undefined,
      "decorates": undefined,
      "decorators": undefined,
      "defaultValue": undefined,
      "extendedBy": undefined,
      "extendedTypes": undefined,
      "flags": Object {},
      "groups": undefined,
      "id": 22,
      "implementationOf": undefined,
      "implementedBy": undefined,
      "implementedTypes": undefined,
      "indexSignature": undefined,
      "inheritedFrom": undefined,
      "kind": 64,
      "kindString": "Function",
      "name": "xxFn",
      "overwrites": undefined,
      "signatures": Array [
        Object {
          "comment": undefined,
          "decorates": undefined,
          "decorators": undefined,
          "flags": Object {},
          "id": 23,
          "implementationOf": undefined,
          "inheritedFrom": undefined,
          "kind": 4096,
          "kindString": "Call signature",
          "name": "xxFn",
          "overwrites": undefined,
          "parameters": undefined,
          "type": Object {
            "name": "void",
            "type": "intrinsic",
          },
          "typeParameter": undefined,
        },
      ],
      "sources": Array [
        Object {
          "character": 16,
          "fileName": "core/math/index.ts",
          "line": 7,
        },
      ],
      "type": undefined,
      "typeParameter": undefined,
    },
    Object {
      "categories": undefined,
      "children": undefined,
      "comment": undefined,
      "decorates": undefined,
      "decorators": undefined,
      "defaultValue": undefined,
      "extendedBy": undefined,
      "extendedTypes": undefined,
      "flags": Object {},
      "groups": undefined,
      "id": 24,
      "implementationOf": undefined,
      "implementedBy": undefined,
      "implementedTypes": undefined,
      "indexSignature": undefined,
      "inheritedFrom": undefined,
      "kind": 64,
      "kindString": "Function",
      "name": "xxFunctionWithMultiSignature",
      "overwrites": undefined,
      "signatures": Array [
        Object {
          "comment": undefined,
          "decorates": undefined,
          "decorators": undefined,
          "flags": Object {},
          "id": 25,
          "implementationOf": undefined,
          "inheritedFrom": undefined,
          "kind": 4096,
          "kindString": "Call signature",
          "name": "xxFunctionWithMultiSignature",
          "overwrites": undefined,
          "parameters": Array [
            Object {
              "comment": undefined,
              "decorates": undefined,
              "decorators": undefined,
              "defaultValue": undefined,
              "flags": Object {},
              "id": 26,
              "kind": 32768,
              "kindString": "Parameter",
              "name": "p",
              "type": Object {
                "name": "string",
                "type": "intrinsic",
              },
            },
          ],
          "type": Object {
            "name": "void",
            "type": "intrinsic",
          },
          "typeParameter": undefined,
        },
        Object {
          "comment": undefined,
          "decorates": undefined,
          "decorators": undefined,
          "flags": Object {},
          "id": 27,
          "implementationOf": undefined,
          "inheritedFrom": undefined,
          "kind": 4096,
          "kindString": "Call signature",
          "name": "xxFunctionWithMultiSignature",
          "overwrites": undefined,
          "parameters": Array [
            Object {
              "comment": undefined,
              "decorates": undefined,
              "decorators": undefined,
              "defaultValue": undefined,
              "flags": Object {},
              "id": 28,
              "kind": 32768,
              "kindString": "Parameter",
              "name": "p",
              "type": Object {
                "name": "number",
                "type": "intrinsic",
              },
            },
          ],
          "type": Object {
            "name": "void",
            "type": "intrinsic",
          },
          "typeParameter": undefined,
        },
      ],
      "sources": Array [
        Object {
          "character": 16,
          "fileName": "core/math/index.ts",
          "line": 9,
        },
      ],
      "type": undefined,
      "typeParameter": undefined,
    },
  ],
  "comment": undefined,
  "decorates": undefined,
  "decorators": undefined,
  "flags": Object {},
  "groups": Array [
    Object {
      "children": Array [
        29,
        30,
      ],
      "kind": 16777216,
      "title": "References",
    },
    Object {
      "children": Array [
        1,
      ],
      "kind": 4,
      "title": "Namespaces",
    },
    Object {
      "children": Array [
        34,
      ],
      "kind": 8,
      "title": "Enumerations",
    },
    Object {
      "children": Array [
        32,
      ],
      "kind": 256,
      "title": "Interfaces",
    },
    Object {
      "children": Array [
        31,
      ],
      "kind": 32,
      "title": "Variables",
    },
    Object {
      "children": Array [
        22,
        24,
      ],
      "kind": 64,
      "title": "Functions",
    },
  ],
  "id": 0,
  "kind": 1,
  "kindString": "Project",
  "name": "typedoc-plugin-cc",
  "originalName": "",
  "sources": Array [
    Object {
      "character": 0,
      "fileName": "index.ts",
      "line": 2,
    },
  ],
}
`;
